### PR TITLE
Fix MD linting in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Rav3nPL**
Format document to consistent markdown rules:

- all lists use "-" (dash) - it was mixed with "*" (asterisk)
- equally indent all lists
- fix two "bare" links
- all headers styles use "atx" - it was mixed with "setext", but we need 3 heading levels there
- removed  needless empty lines 

Added missing comma.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
This is a massive diff, the rendered markdown difference is empty and the travis linter doesn't pass. Slightly tend toward NACK, as it is not worth reviewing.

If you want to contribute to Bitcoin Core, refer to the following section.

## Getting started to contribute to Bitcoin Core

### Setting up your development environment

New developers are very welcome and needed. There are a lot of open issues of any difficulty waiting to be fixed. However, before you start contributing, familiarize yourself with the Bitcoin Core build system and tests. Refer to the documentation in the repository on how to build Bitcoin Core and how to run the unit and functional tests. Once that is done, you are all set.

If you need more help getting started, please refer to the following resources:

* https://medium.com/@amitiu/onboarding-to-bitcoin-core-7c1a83b20365
* https://jonatack.github.io/articles/how-to-review-pull-requests-in-bitcoin-core

### Pick something to work on

If you are looking for useful contributions to help out with, you can

* Search through the [good first issue](https://github.com/bitcoin/bitcoin/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)s or the ones that are [up for grabs](https://github.com/bitcoin/bitcoin/issues?utf8=%E2%9C%93&q=label%3A%22Up+for+grabs%22). Some of them might no longer be applicable. So if you are interested, but unsure, you might want to leave a comment on the issue first.
* Write tests to improve the coverage. Any kind of test is welcome and coverage information can be obtained from a relatively recent [coverage report](https://marcofalke.github.io/btc_cov). If you are unsure, don't hesitate to check back first.
* Help with review and testing. There are easy ones such as the [gui](https://github.com/bitcoin-core/gui/pulls) and [rpc](https://github.com/bitcoin/bitcoin/pulls?q=is%3Apr+is%3Aopen+label%3ARPC%2FREST%2FZMQ). However, review on any open pull request is welcome. Review will also help you understand the codebase better.
* Help on meta projects related to Bitcoin Core, such as a [high-level performance monitor](https://github.com/chaincodelabs/bitcoinperf).
* Join us on irc and let us know what you are interested in.

**Comment by Rav3nPL** [comment]
Diff is big only because of formatting lists.
Rendered markdown is not changed (except for comma) as intended. This is only formatting fix, not contents change.
I have no idea why travis failing, log shows some odd stuff, not related to document itself.

**Comment by hebasto** [comment]
> I have no idea why travis failing, log shows some odd stuff, not related to document itself.

`test/lint/lint-whitespace.sh` -- trailing spaces?

**Comment by jonatack** [comment]
Tend to NACK, this is style change noise for no visible change per [this markdown preview](https://github.com/bitcoin/bitcoin/pull/20086/files?short_path=6a33714#diff-6a3371457528722a734f3c51d9238c13) except for one single comma, and of all the places one could change a comma in the prose, it's not necessarily the only or optimal one.

**Comment by jonatack** [comment]
> Help with review and testing... review on any open pull request is welcome. Review will also help you understand the codebase better.

This is a great suggestion for how to contribute.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,33 +68,33 @@ The developer
...
-  1. Commit patches
+1. Fork repository ([only for the first time](https://help.github.com/en/articles/fork-a-repo))
+2. Create topic branch
+3. Commit patches
 
-For GUI-related issues or pull requests, the https://github.com/bitcoin-core/gui repository should be used.
-For all other issues and pull requests, the https://github.com/bitcoin/bitcoin node repository should be used.
+For GUI-related issues or pull requests, the
+[https://github.com/bitcoin-core/gui](https://github.com/bitcoin-core/gui) repository should be used.
+For all other issues and pull requests, the
```
Comment (hebasto) [inline CONTRIBUTING.md:85]: > @hebasto no trailing spaces there.

```
$ test/lint/lint-whitespace.sh 
This diff appears to have added new lines with trailing whitespace.
The following changes were suspected:

diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
@@ -88,2 +83,4 @@ To contribute a patch, the workflow is as follows:
+For all other issues and pull requests, the
```

**Comment by Rav3nPL** [comment]
@hebasto fixed. IDK how it get there, had no errors when copy/paste from editor.

**Comment by laanwj** [comment]
NACK, this is not worth doing IMO.

**Pull request closed without merging**